# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/FARNHELL/ML-INTERNSHIP/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Lane: Refresh / Content Opportunity Scoring**

This lane focuses on predicting which pieces of content are likely to decline in performance over time. This matters for SEO content management because it lets content teams proactively identify content that may need updates, repurposing, or removal before its performance drops significantly. Catching declining content early means editorial effort goes where it has the most impact, instead of being spread evenly (or spent on content that was already fine).

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Decision:** Which pieces of content should be prioritized for refresh this cycle, out of everything in the catalog.

**Who acts on it:** An SEO content editor / content strategist, who has limited hours per week and needs a ranked shortlist rather than a full audit of every page.

**Action:** The editor pulls the top-scored pages from the queue and schedules them for a content refresh (update stats, re-optimize keywords, expand thin sections, refresh internal links).

Cost of a wrong call:

**False positive** (flagged as declining, isn't really a priority): editor hours spent refreshing content that didn't need it, at the expense of content that did.

**False negative** (truly declining content not flagged): the page keeps losing impressions/sessions unnoticed until the drop is large and harder to reverse.

**Systemic cost:** if the ranking is unreliable, editors stop trusting the queue and go back to manual, ad-hoc triage — the tool loses its value.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [6]:
import pandas as pd

# Construct the raw URL for the CSV file directly from the GitHub repository
csv_url = "https://raw.githubusercontent.com/FARNHELL/ML-INTERNSHIP/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(csv_url)

total = len(df)
trend_counts = df["trend_direction"].value_counts()
declining = trend_counts.get("down", 0)
declining_pct = declining / total * 100

print(f"Total content pieces: {total:,}")
print(f"Declining (trend_direction == 'down'): {declining:,} ({declining_pct:.1f}% of total)")
print()
print("Full trend_direction breakdown:")
print((trend_counts / total * 100).round(1).astype(str) + "%")

Total content pieces: 30,000
Declining (trend_direction == 'down'): 16,262 (54.2% of total)

Full trend_direction breakdown:
trend_direction
down      54.2%
stable    19.9%
up        14.6%
new        7.5%
flat       3.8%
Name: count, dtype: object




**What the numbers say:**

Total content pieces: 30,000
Declining content (trend_direction == 'down'): 16,262 — 54.2% of the catalog
Trend split: decline (54.2%) is more than the other four categories combined (stable 19.9%, up 14.6%, new 7.5%, flat 3.8%)

Over half the catalog is declining at any given time. That's too large a set for an editor to triage by hand every cycle, and too skewed for a coin-flip baseline to be useful — which is exactly the kind of messy-but-real pattern a scoring model earns its place on.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**What I can claim:**

Observed patterns in content performance decline, measured against actual traffic/engagement history

Directional signals — which pages are more likely than others to be declining right now

Decision-support: a ranked shortlist to help an editor prioritize limited refresh hours

**What I cannot claim:**

Causal proof that any feature (word count, freshness, etc.) causes decline — these are associations, not mechanisms

Prediction of Google's algorithm or future rankings — the model reads a snapshot of observed behavior, not


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.